### PR TITLE
Deprecate l2_norm/l2_norm2 in favor of norm/norm2

### PR DESCRIPTION
**Pull request by sdrave**
Following #1066, the implementations of `VectorArray.l2_norm`, `VectorArray.l2_norm2` are moved to private `_norm`, `_norm2` methods. `l2_norm` and `l2_norm2` are deprecated so that there is only one method to compute the Euclidean norm of a `VectorArray`. What that means (the Euclidean norm of the `dofs` of the array) is clarified in the docstring.

Moreover, `almost_equal` has been simplified by replacing the (unused) `norm` parameter by the `sup_norm` parameter.

### CODE REVIEW

File: src/pymor/algorithms/basic.py [hunk 1]
```diff
@@ -34,38 +33,27 @@ def almost_equal(U, V, product=None, norm=None, rtol=1e-14, atol=1e-14):
...
 
-    assert product is None or norm is None
-    assert not isinstance(norm, str) or norm in ('l1', 'l2', 'sup')
-    norm = induced_norm(product) if product is not None else norm
-    if norm is None:
-        norm = 'l2'
-    if isinstance(norm, str):
-        norm_str = norm
-        norm = lambda U: getattr(U, norm_str + '_norm')()
+    assert product is None or not sup_norm
```
Comment (renefritze) [inline src/pymor/algorithms/basic.py:45]: Shouldn't this now be
`assert sup_norm or product is not None` ?

File: src/pymor/vectorarrays/interface.py [hunk 2]
```diff
@@ -407,7 +407,7 @@ def pairwise_inner(self, other, product=None):
 
     @Deprecated(pairwise_inner)
     def pairwise_dot(self, other):
-        return self.pariwise_inner(other)
+        return self.pairwise_inner(other)
```
Comment (renefritze) [inline src/pymor/vectorarrays/interface.py:410]: Seems this was never exercised in tests before, is `pairwise_inner` now?

File: src/pymor/algorithms/basic.py (hunk 1 above)
Comment (HenKlei) [reply src/pymor/algorithms/basic.py:45]: No, I think @sdrave's version is correct.

`assert product is None or not sup_norm` is the same as `assert not (product is not None and sup_norm)` and I think that is what one wants to check here. You do not want that a user provides a product and at the same time sets `sup_norm` to True.

Comment (renefritze) [reply src/pymor/algorithms/basic.py:45]: Alright, If that's the intention then I misread the docstring change.

Comment (HenKlei) [reply src/pymor/algorithms/basic.py:45]: I think the docstring is not optimal.

```
sup_norm
        If specified, use the sup norm to compute the error.
```
sounds like `sup_norm` gets some `norm` as value. Maybe something like
```
sup_norm
        If `True`, use the sup norm to compute the error.
```
fits better.

Comment (sdrave) [reply src/pymor/algorithms/basic.py:45]: Fixed the docstring.

File: src/pymor/vectorarrays/interface.py (hunk 2 above)
Comment (sdrave) [reply src/pymor/vectorarrays/interface.py:410]: No. Maybe it was not so wise to remove the test for `pairwise_dot`. Was too lazy to open a separate PR for this.

**Review by renefritze — APPROVED** [review]

**Comment by sdrave** [comment]
@HenKlei, @pmli, any further comments? I would like to merge this soon, to prevent merge conflicts.

**Review by HenKlei — APPROVED** [review]

**Review by pmli — APPROVED** [review]

**Pull request merged**